uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/absence-tab-custom-work-pattern-modal.js

```diff
@@ -0,0 +1,123 @@
+/* eslint-env amd */
+
+define([
+  'common/moment',
+  'leave-absences/absence-tab/modules/components'
+], function (moment, components) {
+  components.component('absenceTabCustomWorkPatternModal', {
+    bindings: {
+      contactId: '<',
+      dismiss: '&'
```

Comment:
We needed to inject `$uibModalInstance` inside `absenceTabCustomWorkPatternModal` 's controller, but it cannot be done because we are using `v1.3.3` of UI Bootstrap, but this version does not support the angular components. https://stackoverflow.com/a/37381735/2400594

So the only way I found to do this elegantly without updating UI Bootstrap is,
https://github.com/angular-ui/bootstrap/issues/5683#issuecomment-208433560

The same has been used here.
So first I assign the `dismiss`function into a controller variable, 
``` 
controller: ['$uibModalInstance', function ($modalInstance) {
  this.dismiss = $modalInstance.dismiss;
}]
```
Then the same is imported in component
```
bindings: {
  ...
  dismiss: '&'
}